### PR TITLE
Feature #89 ktps 1426 sklearn pipeline monitoring

### PR DESCRIPTION
**Pull request by FrankKr**
Fixes https://github.com/alliander-opensource/short-term-forecasting/issues/89

Concluded that the 'context-specific' steps that used to be in the train_model_core aren't required.
I have left them out, and added a few minor checkpoints on a higher level in the pipeline.

Most work of this story was to include unit tests to test the pipeline:
1) store a new trained model (useful for when the interface of models changes)
2) run the create_forecast_pipeline with only mocking ktpbase-stuff and actually using a previously stored model. Also, here I added a test to check the MAE of the forecast versus the actual load. Perhaps it would make sense to move this test to 'test/regression' or something like that, but for now, this seems Okay. Running the test is very fast, so that is not a reason to move it.

Other minor improvements:
- change Contribution description of workflow from gitflow to github flow
- added some logging on where stuff is stored
- added try/except if old model cannot be compared to new model
- remove ktpbase.logging dependency in taskcontext
- stored a trained model with new interface (e.g. including model.standard_deviation) in unit test folder

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Furthermore the following conventions apply:
 
 ## Git branching
 
-This project uses the [Gitflow Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) and branching model. The `master` branch always contains the latest release, after a release is made new feature branches are branched of `develop`. When a feature is finished it is merged back into `develop`. At the end of a sprint `develop` is merged back into `master` or (optional) into a `release` branch first before it is merged into `master`.
+This project uses the [GitHub flow Workflow](https://guides.github.com/introduction/flow/) and branching model. The `main` branch always contains the latest release. New feature branches are branched from `main`. When a feature is finished it is merged back into `main` via a [Pull Request](https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#:~:text=Pull%20requests%20let%20you%20tell,merged%20into%20the%20base%20branch.).
```
Comment (JanMaartenvanDoorn) [inline CONTRIBUTING.md:38]: Good to update this.

File: openstf/tasks/train_model.py [hunk 2]
```diff
@@ -52,6 +52,8 @@ def train_model_task(pj: dict, context: TaskContext) -> None:
         pj["hyper_params"]["featureset_name"]
     )
 
+    context.checkpoint("Meta-data")
```
Comment (JanMaartenvanDoorn) [inline openstf/tasks/train_model.py:55]: Maybe: "Retrieved meta-data" or "Retrieved prediction job"

File: openstf/tasks/train_model.py [hunk 3]
```diff
@@ -66,6 +68,8 @@ def train_model_task(pj: dict, context: TaskContext) -> None:
         datetime_end=datetime_end,
     )
 
+    context.checkpoint("TimeSeries input")
```
Comment (JanMaartenvanDoorn) [inline openstf/tasks/train_model.py:71]: Maybe "Retrieved timeseries input" is better here as indicates more accurately where we are in the procces.

File: openstf/pipeline/train_model.py [hunk 4]
```diff
@@ -159,25 +159,26 @@ def train_model_pipeline_core(
...
-    if old_model is not None:
+    if old_model:
         combined = train_data.append(validation_data)
         x_data, y_data = combined.iloc[:, 1:], combined.iloc[:, 0]
 
         # Score method always returns R^2
         score_new_model = model.score(x_data, y_data)
-        score_old_model = old_model.score(x_data, y_data)
+        try:
+            score_old_model = old_model.score(x_data, y_data)
```
Comment (JanMaartenvanDoorn) [inline openstf/pipeline/train_model.py:169]: Just being curious: Why is the try, catch  required here?

**Review by JanMaartenvanDoorn — APPROVED** [review]
Nice additions! Some minor comments regarding the phrasing of the check points.

File: openstf/tasks/train_model.py (hunk 2 above)
Comment (FrankKr) [reply openstf/tasks/train_model.py:55]: good suggestion, I'll add it

File: openstf/tasks/train_model.py (hunk 3 above)
Comment (FrankKr) [reply openstf/tasks/train_model.py:71]: Good suggestion, I'll add it

File: openstf/pipeline/train_model.py (hunk 4 above)
Comment (FrankKr) [reply openstf/pipeline/train_model.py:169]: if the old model has different feature names compared to the new model, I'll add it as a comment

**Pull request merged**